server/services/featureFlagService.test.ts

```diff
@@ -0,0 +1,57 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import retry from 'jest-retries'
+import LoggedInUser from '../models/loggedInUser'
+import FeatureFlagService from './featureFlagService'
+
+describe('featureFlagService', () => {
+  describe('enableForUser', () => {
+    const makeRandomUserIds = (howMany = 1000) => {
+      return Array.from({ length: howMany }, () => {
+        return {
+          userId: Math.random()
+            .toString(36)
+            .replace(/[^a-z]+/g, ''),
+        } as LoggedInUser
+      })
+    }
+
+    it('always returns false for 0 percent', () => {
+      makeRandomUserIds().forEach(it => {
+        expect(FeatureFlagService.enableForUser(it, 0)).toEqual(false)
+      })
+    })
+
+    it('always returns true for 100 percent', () => {
+      makeRandomUserIds().forEach(it => {
+        expect(FeatureFlagService.enableForUser(it, 100)).toEqual(true)
+      })
+    })
+
+    retry('returns roughly the right number', 10, () => {
```

Comment:
Not sure there is a better way to test this, but is there still a chance these could fail? I see you've given a bit of leeway.

Reply:
updated the test so that it users a real sample of user ids and validates a single number.